patches/com/mojang/blaze3d/pipeline/RenderPipeline.java.patch

```diff
@@ -41,9 +96,93 @@
 +        builder.vertexFormatMode = Optional.of(this.vertexFormatMode);
 +        builder.depthBiasScaleFactor = this.depthBiasScaleFactor;
 +        builder.depthBiasConstant = this.depthBiasConstant;
++        builder.stencilTest = this.stencilTest;
 +        return builder;
 +    }
 +
      public static RenderPipeline.Builder builder(RenderPipeline.Snippet... p_405781_) {
          RenderPipeline.Builder renderpipeline$builder = new RenderPipeline.Builder();
  
+@@ -208,6 +_,7 @@
+         private Optional<VertexFormat.Mode> vertexFormatMode = Optional.empty();
+         private float depthBiasScaleFactor;
+         private float depthBiasConstant;
++        private Optional<net.neoforged.neoforge.client.stencil.StencilTest> stencilTest = Optional.empty();
+ 
+         Builder() {
+         }
+@@ -364,6 +_,11 @@
+             return this;
+         }
+ 
++        public RenderPipeline.Builder withStencilTest(net.neoforged.neoforge.client.stencil.StencilTest stencilTest) {
++            this.stencilTest = Optional.of(stencilTest);
++            return this;
++        }
++
```

Comment:
Note that this makes it impossible to disable the stencil test. Do we want to add `withoutStencilTest()` as well? (Following `withoutBlend` above.

Reply:
To allow pipeline modifiers to disable stencil without forcing use of a no-op StencilTest (which can be done), probably.